doc/contributing.rst

```diff
@@ -14,6 +14,115 @@ If your contributions will make use of parallel backends for using more than
 one core, please see the additional installation steps in our
 :doc:`parallel backend guide <parallel>`.
 
+Overview of contribution process
```

Comment:
I think if you use some of the wording from this section to make the next section a little more verbose, one could get rid of the section altogether.